pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -542,23 +542,12 @@ pub mod pallet {
 				return (Err(Error::<T>::RouterNotFound.into()), read_weight);
 			};
 
-			let serialized = message.serialize();
-			let serialized_len = serialized.len() as u64;
-
-			// TODO: do we really need to return the weights in send() if later we use the
-			// defensive ones?
-			let (result, router_weight) = match router.send(sender, serialized) {
+			let (result, router_weight) = match router.send(sender, message.serialize()) {
 				Ok(dispatch_info) => (Ok(()), dispatch_info.actual_weight),
 				Err(e) => (Err(e.error), e.post_info.actual_weight),
 			};
 
-			(
-				result,
-				router_weight
-					.unwrap_or(Weight::from_parts(LP_DEFENSIVE_WEIGHT_REF_TIME, 0))
-					.saturating_add(read_weight)
-					.saturating_add(Weight::from_parts(0, serialized_len)),
-			)
```

Comment:
After investigating this. We delegate the processing weight to the router, which should take care of the bytes sent. In case we need some extra weight, it should be added to the router.

Reply:
BTW, We need to check the current `XcmDomain::overall_weight` used in production to be sure it's higher than our needs